### ISSUE TITLE
Generated java for polymorphic union literals is incorrect.

### ISSUE THREAD

**Issue by timbod7** [timbod7/adl#86]
With a declaration like this:

```
union Loading<T> {
  T value;
  Void loading;
  String error;
};
struct LoadingUser {
  Loading<WithDbId<AppUser>> user = "loading";
};
```

Literals are being create with the type name like this:

```
    this.user = Loading<WithDbId<ExportDesignObjectsReq>>.loading()
```

But rather should use the class name like this:

```
    this.user = Loading.loading();
```

**Comment by timbod7** [comment]
Fix needed here I think:

https://github.com/timbod7/adl/blob/master/haskell/compiler/src/ADL/Compiler/Backends/Java/Internal.hs#L341

**Comment by timbod7** [comment]
fixed with b0123f6
